exercises/practice/all-your-base/.approaches/from-base-ten/content.md

```diff
@@ -0,0 +1,118 @@
+# Converting from base 10
+
+There are different strategies to convert a decimal number to another base.
+Here is a selection of a few approaches and thoughts on their performance.
+Keep in mind, that the number of digits you need to process is the most important factor for your algorithms performance.
+A simple for loop can be a lot faster for a small number than a [parallel algorithm][parallel-computing] that has more overhead to be set up.
+
+
+## 1: While-loop with `%` and `/`
+
+This approach uses a [while-loop][while-loop] to repeatedly divide the `intermediate` by the `output_base`, extracting the [remainders][remainder-operator]c as digits which are then stored in the `output_digits` vector.
+
+```cpp
+std::vector <unsigned int> output_digits{};
+while (intermediate > 0) {
+    output_digits.emplace_back(intermediate % output_base);
+    intermediate /= output_base;
+}
+```
+
+The approach is simple, straightforward, and easy to understand.
+The order of operation makes it necessary to revert the final `vector`.
+As the loop constantly writes to the `intermediate` variable and needs to do so in order, the approach is strictly serial.
+
+## 2: For-loop with `%` and `/`
+
+Instead of the while-loop from the approach above, you can also use a [for-loop][for-loop].
+
+```cpp
+std::vector <unsigned int> output_digits{};
+for (; intermediate > 0; intermediate /= output_base) {
+    output_digits.emplace_back(intermediate % output_base);
+}
+```
+
+The result is a bit more concise, but has the same properties as the while version.
+
+## 3: Using `div`
+
+When you need both the remainder and the quotient from a division operation, you can use [`std::div`][div-function].
+
+```cpp
+std::vector <unsigned int> output_digits{};
+std::ldiv_t v{intermediate, 0};        
+while (v.quot > 0) {
+    v = std::div(v.quot, output_base);
+    output_digits.emplace_back(v.rem);
+}
+```
+
+Depending on the processor this might save one operation per loop iteration.
+You do need to allocate some memory for the `v` variable of type `std::ldiv_t`.
```

Comment:
I wouldn't call that "allocate" because that sounds too much like "dynamically allocated memory".
How about something like "define a variable".